### ISSUE TITLE
Feature Request : null auto casting to any cpp.Pointer

### ISSUE THREAD

**Issue by delahee** [HaxeFoundation/hxcpp#380]
Hi @hughsando !

TLDR : Can you make something to relax null transtyping to any cpp.Pointer<...>

Long version : 
I have been experimenting with hxcpp cpp.Pointer and stumbled upon how to create the good old nullptr...

var pint = 0;
//nope
//var p : cpp.Pointer<Int> = cpp.Pointer.addressOf( null );

//nope
//var p : cpp.Pointer<Int> = cpp.Pointer.addressOf( 0 );

//nope
//var p : cpp.Pointer<Int> = cpp.Pointer.fromRaw( 0 );

//nope
//var p : cpp.Pointer<Int> = new cpp.Pointer(0);

//yep but not what I want
//var p : cpp.Pointer<Int> = cpp.Pointer.addressOf( pint );

//ok 
var p : cpp.Pointer<Int> = null;

//ok 
var p : cpp.Pointer<Int> = cast null;

//ok 
var p = null;

foo("myfile", p); //ok in anyway we put

//ARRGHH
foo("myfile", null ); //cannot convert parameter 1 from 'null' to 'Int *'

That is this case we could improve a lot and then avoid
var a=null,b=null,c=null in many places for complex bindings...

Thanks !

**Comment by hughsando** [comment]
Null now casts to any pointer, so this should work.

```
using cpp.NativeString;

extern class Extern
{
   @:native("printString")
   public static function print(value:cpp.ConstPointer< cpp.Char >) : Void { }
}

@:cppFileCode('
void printString(const char *s)
{
   printf("String : %s\\n",s);
}
')
class Test {
  static function main() {
     Extern.print( "hello".c_str() );
     Extern.print( null  );
  }
}
```
